utils/tappable.go

```diff
@@ -0,0 +1,55 @@
+package utils
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/widget"
+)
+
+var _ fyne.Widget = (*tappableObject)(nil)
+var _ fyne.Tappable = (*tappableObject)(nil)
+
+// handles the tap event.
+type tappableObject struct {
+	widget.BaseWidget
+	object   fyne.CanvasObject
+	onTapped func(*fyne.PointEvent)
+}
+
+// Content returns the encapsulated widget.
+func (t *tappableObject) Content() fyne.CanvasObject {
+	return t.object
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer.
+//
+// Implements: fyne.Widget
+func (t *tappableObject) CreateRenderer() fyne.WidgetRenderer {
+	if t.object == nil {
+		return nil
+	}
+	if o, ok := t.object.(fyne.Widget); ok {
+		return o.CreateRenderer()
+	}
+	return widget.NewSimpleRenderer(t.object)
+}
+
+// Tapped reacts on tap (or click) events.
+//
+// Implements: fyne.Tappable
+func (t *tappableObject) Tapped(e *fyne.PointEvent) {
+	if t.object == nil {
+		return
+	}
+
+	if o, ok := t.object.(fyne.Tappable); ok {
+		o.Tapped(e)
+	}
+	t.onTapped(e)
+}
+
+// SetTappable set the object to be tappable.
+func SetTappable(object fyne.CanvasObject, ontapped func(*fyne.PointEvent)) fyne.CanvasObject {
```

Comment:
Actually, because I accept `fyne.CanvasObject` as parameter, I thought that was not the right place. And because we can have a lot of wrappers (that could be a good name ;) ) that was a "solution" to separate these functions.

For the cast-like `Tappable` I'm afraid that the users could have confused with the `fyne.Tappable` interface.

`TappableWidget` has got the same problem as what I say on first, I mean that I accept any CanvasObject, so is it correct to only speak about widgets?

Reply:
A pretty good name, and maybe this discussion indicates that `wrapper` does make sense for the package. Pretty generic, but not an incorrect name ;)

Reply:
That's OK for me, I will rewrite this ;)

Reply:
Changed to `wrapper` – tests added.
Sorry for the delay.

Reply:
> Wondering if `fyne.MakeTappable` could make sense too.

I do like this more than the `SetTappable` form as it's not really a setter...